pulp_rpm/app/tasks/publishing.py

```diff
@@ -117,18 +117,51 @@ def publish_artifacts(self, content, prefix=""):
             content__pulp_type=Package.get_pulp_type()
         )
 
+        paths = set()
+        duplicated_paths = []
         for content_artifact in contentartifact_qs.values("pk", "relative_path").iterator():
             relative_path = content_artifact["relative_path"]
             relative_path = os.path.join(
                 prefix, PACKAGES_DIRECTORY, relative_path.lower()[0], relative_path
             )
+            #
+            # Some Suboptimal Repos have the 'same' artifact living in multiple places.
+            # Specifically, the same NEVRA, in more than once place, **with different checksums**
+            # (since if all that was different was location_href there would be only one
+            # ContentArtifact in the first place).
+            #
+            # pulp_rpm wants to publish a 'canonical' repository-layout, under which an RPM
+            # "name-version-release-arch" appears at "Packages/n/name-version-release-arch.rpm".
+            # Because the assumption is that Packages don't "own" their path, only the filename
+            # is kept as relative_path.
+            #
+            # In this case, we have to pick one - which is essentially what the rest of the RPM
+            # Ecosystem does when faced with the impossible. This code takes the first-found. We
+            # could implement something more complicated, if there are better options
+            # (choose by last-created maybe?)
+            #
+            # Note that this impacts only actual publications, which produce the "standard" RPM
```

Comment:
I would say "only impacts user-created publications", and below "A publication created by mirror-sync retains whatever layout..."

Reply:
I like it, done.